// openvidu-node-client/lib/Archive.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var Archive = /** @class */ (function () {
    function Archive(json) {
        this.size = 0;
        this.duration = 0;
        this.hasaudio = true;
        this.hasvideo = true;
        this.id = json['id'];
        this.name = json['name'];
        this.sessionId = json['sessionId'];
        this.createdAt = json['createdAt'];
        this.size = json['size'];
        this.duration = json['duration'];
        this.url = json['url'];
        this.hasaudio = json['hasAudio'];
        this.hasvideo = json['hasVideo'];
        this.status = json['status'];
    }
    Archive.prototype.getStatus = function () {
        return this.status;
    };
    Archive.prototype.getId = function () {
        return this.id;
    };
    Archive.prototype.getName = function () {
        return this.name;
    };
    Archive.prototype.getSessionId = function () {
        return this.sessionId;
    };
    Archive.prototype.getCreatedAt = function () {
        return this.createdAt;
    };
    Archive.prototype.getSize = function () {
        return this.size;
    };
    Archive.prototype.getDuration = function () {
        return this.duration;
    };
    Archive.prototype.getUrl = function () {
        return this.url;
    };
    Archive.prototype.hasAudio = function () {
        return this.hasaudio;
    };
    Archive.prototype.hasVideo = function () {
        return this.hasvideo;
    };
    return Archive;
}());
exports.Archive = Archive;
(function (Archive) {
    var Status;
    (function (Status) {
        Status[Status["starting"] = 0] = "starting";
        Status[Status["started"] = 1] = "started";
        Status[Status["stopped"] = 2] = "stopped";
        Status[Status["available"] = 3] = "available";
        // stopped recordings if property 'openvidu.recording.free-access' is true
        Status[Status["failed"] = 4] = "failed"; // The recording has failed
    })(Status = Archive.Status || (Archive.Status = {}));
})(Archive = exports.Archive || (exports.Archive = {}));
exports.Archive = Archive;
